Utils/src/FastSimWeightPR31285To36122.cc

```diff
@@ -0,0 +1,202 @@
+    const int lowerBoundForLightHadron = 100;
+    const int lowerBoundForCHadron = 400;
+    const int lowerBoundForBHadron = 500;
+    const int upperBoundForHadron = 600;
+    for(const auto& genJet : *genJets) 
+    {
+           if (!(genJet.pt()>genJetPtThreshold)) continue;
+           jetHasOffendingGp_ = false;//must use
+           mindr = 99;
+           hadronFlavor = -11;
+           leadHadronPt = 1;
+           originxy_squared = -1;
+           for(const auto& recJet : *recJets) 
+            {
+                dr = deltaR(recJet.p4(),genJet.p4());
+                if (!(dr<drThreshold && dr<mindr)) continue;
+                mindr = dr;
+                hadronFlavor = std::max(1,recJet.hadronFlavour());
+                for(const auto& iPart : *genPartCands) {
+                    if (!(iPart.pt()>genParticlePtThreshold)) continue;
+                    pdgid = abs(iPart.pdgId());
+                    if (!(deltaR(iPart,genJet)<drThreshold)) continue;
+                    if(!(pdgid>lowerBoundForLightHadron && pdgid<upperBoundForHadron)) continue; //only consider hadrons
+                    if (iPart.pt()>leadHadronPt) {
+                        if ( (hadronFlavor<4 && pdgid>lowerBoundForLightHadron && pdgid<lowerBoundForCHadron) ||//only consider hadrons
+                             (hadronFlavor==4 && pdgid>lowerBoundForCHadron && pdgid<lowerBoundForBHadron) ||//only consider hadrons
+                             (hadronFlavor==5 && pdgid>lowerBoundForCHadron && pdgid<upperBoundForHadron) )//only consider hadrons
+                        {
+                            leadHadronPt = iPart.pt();
+                        }
+                    }                       
+                    if(!(iPart.numberOfDaughters()>1)) continue;
+                    originxy_squared = pow(iPart.vx(),2) + pow(iPart.vy(),2);
+                    if(!(originxy_squared<justInsidePipe_squared)) continue;
+                    decayxy_squared = pow(iPart.daughter(0)->vx(),2) + pow(iPart.daughter(0)->vy(),2);
+                    if (decayxy_squared>justOutsidePipe_squared and decayxy_squared<farOutsideCaloButNotTooFarAway_squared)
+                    {
+                        jetHasOffendingGp_ = true;
+                        break;
+                    }
+                }
+            }
+            leadHadronPt = xax->GetBinLowEdge(xax->FindBin(std::min(leadHadronPt, justInsideTheHighestBinEdge)));
+            if (jetHasOffendingGp_)
+            {
+                EventWeight*=0;
+            }
+            else
+            {
+               if (hadronFlavor<4) EventWeight*=hRatio_GenJetHadronPtGenJetHadronFlavorLt4->Interpolate(leadHadronPt);
+               else if (hadronFlavor==4) EventWeight*=hRatio_GenJetHadronPtGenJetHadronFlavorEqEq4->Interpolate(leadHadronPt);
+               else if (hadronFlavor==5) EventWeight*=hRatio_GenJetHadronPtGenJetHadronFlavorEqEq5->Interpolate(leadHadronPt);           
+            }
+        }
```

Comment:
the overall indentation here is wrong: this last bracket should line up with the bracket that starts the for loop

Reply:
should be better now